src/cloudai/_core/installables.py

```
@@ -119,6 +120,44 @@ def repo_name(self) -> str:
+            return False, "Cloned repo has some submodules initialized but requires none to be."
+
+        return True, ""
+
+    def ensure_submodules_state(self, repo_path: Path) -> tuple[bool, str]:
+        """Ensure submodules state in the cloned repo matches self.init_submodules (install or deinstall them)."""
+        submodules_are_ok, _ = self.check_submodules_state(repo_path)
+        if submodules_are_ok:
+            return True, ""
+
+        cmd = ["update", "--init", "--recursive"] if self.init_submodules else ["deinit", "--all", "--force"]
+        result = subprocess.run(["git", "submodule", *cmd], cwd=str(repo_path), capture_output=True, text=True)
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)